File: 0791-custom-sort-string/0791-custom-sort-string.cpp
class Solution {
public:
    string customSortString(string order, string s) {
        vector<int>freq(26, 0);
        
        for(char i : s)   
            ++freq[i - 'a'];
        
        string res;
        
        for(char i : order){
            while(freq[i - 'a'] != 0){
                res.push_back(i);
                --freq[i - 'a'];
            }
        }
        
        for(int i = 0; i < 26; i++){
            while(freq[i] != 0){
                res.push_back((char)(i + 'a'));
                --freq[i];
            }
        }
        
        return res;
    }
};

File: 0791-custom-sort-string/0791-custom-sort-string.java
class Solution {
    public String customSortString(String order, String s) {
        int[] freq = new int[26];
        
        for(char i : s.toCharArray())   
            ++freq[i - 'a'];
        
        StringBuilder sb = new StringBuilder();
        
        for(char i : order.toCharArray()){
            while(freq[i - 'a']-- > 0){
                sb.append(i);
            }
        }
        
        for(int i = 0; i < 26; i++){
            while(freq[i]-- > 0){
                sb.append((char)(i + 'a'));
            }
        }
        
        return sb.toString();
    }
}